Update geo-inference parameters

mpelchat04 opened the pull request:
# Description
This PR updates the parameters from geo-inference (workers, prediction_threshold, patch_size).  
It also reduces the size of the docker image created.  


## Type of change
What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other 

## How Has This Been Tested?  
- Inference integration test  
- On HPC, on multiple images using the image `tst_gdl_3_1.sif` 


# Checklist:
If you're unsure about any of these, don't hesitate to ask. We're here to help! 
- [X ] My code follows the code style of this project
- [X ] I have performed a self-review of my own code
- [X ] My PR has a descriptive title
- [X ] My code follows PEP8 (most of it)
- [X ] My code have the proper documentation in the code
- [X ] I have made corresponding changes to the documentation (if applicable)
- [X ] My changes generate no new errors
- [X ] My code is covered by the existing tests (if applicable)
- [ ] My code includes tests that show that my feature works
- [ ] My PR passes Github Actions

ms888ekb commented on Dockerfile:
```diff
@@ -1,46 +1,38 @@
 RUN apt-get update \
     && apt-get install -y --no-install-recommends git wget unzip bzip2 build-essential sudo \
-    && apt-key del 7fa2af80 \
-    && wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/cuda-keyring_1.0-1_all.deb \
```
dont we need these cuda keyring stuff anymore?

mpelchat04 replied:
Nope. Tested with a local docker image, the CI test image and on HPC. All works fine without these.

ms888ekb commented on inference_segmentation.py:
```diff
@@ -1,12 +1,14 @@
 from typing import Dict, Sequence, Union
 from dataset.stacitem import SingleBandItemEO
+import rasterio
+from shutil import move
```
can we start organizing module loading according to PEP8? i.e. system libraries in the first block, external in the second, locally written libraries in the third.

mpelchat04 replied:
Changed it in this script.  
Following the refactoring of all the other modules, this will be done in the other module as well.

ms888ekb commented on inference_segmentation.py:
```diff
@@ -1,15 +1,16 @@
 import csv
-from math import sqrt
+import rasterio
+
```
it should look like this:

```
from shutil import move
from pathlib import Path
from numbers import Number
from tempfile import mkstemp
from typing import Dict, Sequence, Union

from tqdm import tqdm
import csv
import rasterio
from omegaconf import DictConfig

from utils.aoiutils import aois_from_csv
from dataset.stacitem import SingleBandItemEO
from utils.logger import get_logger, set_tracker
from geo_inference.geo_inference import GeoInference
from utils.utils import get_device_ids, get_key_def, set_device
```

but I think that PEP8 refactoring can be done later in a separate PR.

mpelchat04 commented:
@CharlesAuthier and @ms888ekb I added the support for Test-Time augmentation and corrected the model for the tests to pass.  

Final review, I swear. :)

ms888ekb commented on environment.yml:
```diff
@@ -1,16 +1,27 @@
+  - pystac>=1.10.1
   - pytest>=7.1
+  - python>=3.11
+  - pytorch>=2.3
```
why do we need pytorch and pytorch-cuda at the same time? pytorch-cuda can work on CPU if needed. Won't having two different versions cause conflicts?

mpelchat04 replied:
It is the recommended way to install pytorch in a conda env.: https://pytorch.org/
